zzz_generated/app-common/app-common/zap-generated/attribute-type.cpp

```diff
@@ -0,0 +1,185 @@
+        return 5;
+    case ZCL_INT48S_ATTRIBUTE_TYPE:
+        return 6;
+    case ZCL_INT56S_ATTRIBUTE_TYPE:
+        return 7;
+    case ZCL_INT64S_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_ENUM8_ATTRIBUTE_TYPE:
+        return 1;
+    case ZCL_ENUM16_ATTRIBUTE_TYPE:
+        return 2;
+    case ZCL_PRIORITY_ATTRIBUTE_TYPE:
+        return 1;
+    case ZCL_STATUS_ATTRIBUTE_TYPE:
+        return 1;
+    case ZCL_SINGLE_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_DOUBLE_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_OCTET_STRING_ATTRIBUTE_TYPE:
+        return 0;
+    case ZCL_CHAR_STRING_ATTRIBUTE_TYPE:
+        return 0;
+    case ZCL_LONG_OCTET_STRING_ATTRIBUTE_TYPE:
+        return 0;
+    case ZCL_LONG_CHAR_STRING_ATTRIBUTE_TYPE:
+        return 0;
+    case ZCL_ARRAY_ATTRIBUTE_TYPE:
+        return 0;
+    case ZCL_STRUCT_ATTRIBUTE_TYPE:
+        return 0;
+    case ZCL_GROUP_ID_ATTRIBUTE_TYPE:
+        return 2;
+    case ZCL_ENDPOINT_NO_ATTRIBUTE_TYPE:
+        return 2;
+    case ZCL_VENDOR_ID_ATTRIBUTE_TYPE:
+        return 2;
+    case ZCL_DEVTYPE_ID_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_FABRIC_ID_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_FABRIC_IDX_ATTRIBUTE_TYPE:
+        return 1;
+    case ZCL_ENTRY_IDX_ATTRIBUTE_TYPE:
+        return 2;
+    case ZCL_DATA_VER_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_EVENT_NO_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_SEMTAG_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_NAMESPACE_ATTRIBUTE_TYPE:
+        return 1;
+    case ZCL_TAG_ATTRIBUTE_TYPE:
+        return 1;
+    case ZCL_SYSTIME_US_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_SYSTIME_MS_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_ELAPSED_S_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_TEMPERATURE_ATTRIBUTE_TYPE:
+        return 2;
+    case ZCL_POWER_MW_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_AMPERAGE_MA_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_VOLTAGE_MV_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_ENERGY_MWH_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_TOD_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_DATE_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_EPOCH_US_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_EPOCH_S_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_POSIX_MS_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_PERCENT_ATTRIBUTE_TYPE:
+        return 1;
+    case ZCL_PERCENT100THS_ATTRIBUTE_TYPE:
+        return 2;
+    case ZCL_CLUSTER_ID_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_ATTRIB_ID_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_FIELD_ID_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_EVENT_ID_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_COMMAND_ID_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_ACTION_ID_ATTRIBUTE_TYPE:
+        return 1;
+    case ZCL_TRANS_ID_ATTRIBUTE_TYPE:
+        return 4;
+    case ZCL_NODE_ID_ATTRIBUTE_TYPE:
+        return 8;
+    case ZCL_IPADR_ATTRIBUTE_TYPE:
```

Comment:
All the zero sized ones are odd

Reply:
Zero means "the size is not determined by just knowing the type".